mapbox_vector_tile/__init__.py

```diff
@@ -1,25 +1,36 @@
+from typing import Dict, Optional
+
 from mapbox_vector_tile import decoder, encoder
+from mapbox_vector_tile.options import DEFAULT_DECODE_OPTIONS, DEFAULT_ENCODE_OPTIONS
+from mapbox_vector_tile.typing import DecodeOptionsType, EncodeOptionsType
 
 
-def decode(tile, y_coord_down=False):
-    vector_tile = decoder.TileData()
-    message = vector_tile.get_message(tile, y_coord_down)
+def decode(
+    tile,
+    layer_options: Optional[Dict[str, DecodeOptionsType]] = None,
+    default_options: DecodeOptionsType = DEFAULT_DECODE_OPTIONS,
+):
+    vector_tile = decoder.TileData(pbf_data=tile, options=layer_options, default_options=default_options)
+    message = vector_tile.get_message()
     return message
 
 
 def encode(
     layers,
-    quantize_bounds=None,
-    y_coord_down=False,
-    extents=4096,
-    on_invalid_geometry=None,
-    check_winding_order=True,
+    options: Optional[Dict[str, EncodeOptionsType]] = None,
+    default_options: EncodeOptionsType = DEFAULT_ENCODE_OPTIONS,
 ):
-    vector_tile = encoder.VectorTile(extents, on_invalid_geometry, check_winding_order=check_winding_order)
+    vector_tile = encoder.VectorTile(default_options=default_options)
+    if options is None:
+        options = dict()
     if isinstance(layers, list):
         for layer in layers:
-            vector_tile.add_features(layer["features"], layer["name"], quantize_bounds, y_coord_down)
+            layer_name = layer["name"]
+            layer_options = options.get(layer_name, None)
+            vector_tile.add_layer(features=layer["features"], name=layer_name, options=layer_options)
```

Comment:
There is an `options` parameter for the method `add_layer` to override the options given to the constructor.